stl/inc/deque

```diff
@@ -1473,47 +1473,34 @@ private:
 
 public:
     void pop_front() noexcept /* strengthened */ {
+#if _MSVC_STL_HARDENING_DEQUE || _ITERATOR_DEBUG_LEVEL != 0
+        _STL_VERIFY(!empty(), "pop_front() called on empty deque");
+#endif
+
 #if _ITERATOR_DEBUG_LEVEL == 2
-        if (empty()) {
-            _STL_REPORT_ERROR("pop_front() called on empty deque");
-        } else { // something to erase, do it
-            _Orphan_off(_Myoff());
-            _Alty_traits::destroy(_Getal(), _Get_data()._Address_subscript(_Myoff()));
-            if (--_Mysize() == 0) {
-                _Myoff() = 0;
-            } else {
-                ++_Myoff();
-            }
-        }
-#else // ^^^ _ITERATOR_DEBUG_LEVEL == 2 / _ITERATOR_DEBUG_LEVEL < 2 vvv
+        _Orphan_off(_Myoff());
+#endif
         _Alty_traits::destroy(_Getal(), _Get_data()._Address_subscript(_Myoff()));
         if (--_Mysize() == 0) {
             _Myoff() = 0;
         } else {
             ++_Myoff();
         }
-#endif // ^^^ _ITERATOR_DEBUG_LEVEL < 2 ^^^
     }
 
     void pop_back() noexcept /* strengthened */ {
-#if _ITERATOR_DEBUG_LEVEL == 2
-        if (empty()) {
-            _STL_REPORT_ERROR("pop_back() called on empty deque");
-        } else { // something to erase, do it
-            size_type _Newoff = _Myoff() + _Mysize() - 1;
-            _Orphan_off(_Newoff);
-            _Alty_traits::destroy(_Getal(), _Get_data()._Address_subscript(_Newoff));
-            if (--_Mysize() == 0) {
-                _Myoff() = 0;
-            }
-        }
-#else // ^^^ _ITERATOR_DEBUG_LEVEL == 2 / _ITERATOR_DEBUG_LEVEL < 2 vvv
+#if _MSVC_STL_HARDENING_DEQUE || _ITERATOR_DEBUG_LEVEL != 0
+        _STL_VERIFY(!empty(), "pop_back() called on empty deque");
+#endif
+
         size_type _Newoff = _Myoff() + _Mysize() - 1;
+#if _ITERATOR_DEBUG_LEVEL == 2

```

Comment:
Same question as above

Reply:
Same answer - `IDL == 2` guards the ABI-breaking machinery for iterator invalidation.